### PR TITLE
Update provider dependencies

### PR DESCRIPTION
**Pull request by JoelSpeed**
While trying to run a spot VM on Azure, I noticed that the defaulting webhook was removing the spot vm options from the providerspec. This is because the dependencies are out of date and it was not aware of the new field.

This PR bumps the Azure, AWS and GCP provider dependencies to their master head as it is today. This should include any new fields that have been added recently and mean that the webhooks stop removing fields!

### CODE REVIEW

File: service.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: Service
```
Comment (enxebre) [inline service.yaml:2]: I think this got here by accident?

File: simulate-termination.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
+apiVersion: batch/v1
```
Comment (enxebre) [inline simulate-termination.yaml:1]: I think this got here by accident?

File: service.yaml (hunk 1 above)
Comment (JoelSpeed) [reply service.yaml:2]: This is the second PR I've done this with, I really need to delete those files 🤦

**Comment by enxebre** [comment]
> I noticed that the defaulting webhook was removing the spot vm options from the providerspec

Was it because of unmarshalling to the type with out the field?

/approve

**Comment by JoelSpeed** [comment]
> Was it because of unmarshalling to the type with out the field?

That's exactly it yes, unmarshaling the raw data during the defaulting loses any fields that aren't present in the struct with the version of the vendor. Every time we add a new field to on of the provider configs we will need to revendor.

**Pull request merged**